src/d/snd/d_snd_3d_manager.cpp

```diff
@@ -0,0 +1,137 @@
+#include "d/snd/d_snd_3d_manager.h"
+
+#include "d/snd/d_snd_3d_engine.h"
+#include "d/snd/d_snd_mgr.h"
+#include "egg/math/eggMatrix.h"
+#include "egg/math/eggVector.h"
+#include "nw4r/math/math_arithmetic.h"
+#include "nw4r/math/math_types.h"
+#include "nw4r/snd/snd_SoundArchive.h"
+#include "toBeSorted/music_mgrs.h"
+
+SND_DISPOSER_DEFINE(dSnd3DManager_c);
+
+dSnd3DManager_c::dSnd3DManager_c() : mIsSetup(false), mCalculationsFresh(0) {
+    mCameraPosSqVelocity = 0.0f;
+    mCameraAtSqVelocity = 0.0f;
+    mCamDistance = 0.9f;
+    mTimer = 0;
+    mpEngine = new dSnd3DEngine_c();
+    mCameraDirectionNormalized.x = 0.0f;
+    mCameraDirectionNormalized.y = 0.0f;
+    mCameraDirectionNormalized.z = 1.0f;
+}
+
+void dSnd3DManager_c::setup() {
+    if (mIsSetup) {
+        return;
+    }
+
+    nw4r::snd::SoundArchive *archive = dSndMgr_c::GetInstance()->getArchive();
+    u32 requiredSize = mManager.GetRequiredMemSize(archive);
+    void *buf = dSndMgr_c::GetInstance()->getSoundHeap()->Alloc(requiredSize);
+    mManager.Setup(archive, buf, requiredSize);
+    mManager.SetEngine(mpEngine);
+    mManager.SetMaxPriorityReduction(32);
+    mManager.SetField0x20(0.9f);
+    mManager.SetField0x24(3400.0f / 3.0f);
+    mListener.SetMaxVolumeDistance(300.0f);
+    mListener.SetUnitDistance(1000.0f);
+    mListener.SetInteriorSize(400.0f);
+    mManager.SetBiquadFilterType(3);
+    mListener.SetUnitBiquadFilterMax(0.5f);
+    mListener.SetUnitBiquadFilterValue(0.2f);
+    mManager.GetListenerList().Insert(mManager.GetListenerList().GetEndIter(), &mListener);
+    mIsSetup = true;
+}
+
+void dSnd3DManager_c::resetCamDistance() {
+    mCamDistance = 0.9f;
+}
+
+void dSnd3DManager_c::setCamDistance(f32 value) {
+    // @bug ? not actually clamped
+    (void)nw4r::ut::Clamp(value, -1.0f, 2.0f);
```

Comment:
consistency with other clamp bugs
```suggestion
    nw4r::ut::Clamp(value, -1.0f, 2.0f);
```